app/controllers/application_controller.rb

```diff
@@ -38,6 +38,7 @@ def current_role
 
     @role = Role.find_by(id: session[:current_role]) || UsersRole.current_role_for(current_user)
     session[:current_role] = @role&.id
+    UsersRole.set_last_role_for(current_user, @role)
```

Comment:
Do we want to do this every time we access `current_role`? That would happen on pretty much every request. Do we even need to set the session variable here? Seems like we already set it on login and switch.

Reply:
You are correct. I was just testing poorly.